test/jasmine/tests/hover_label_test.js

```diff
@@ -501,6 +501,128 @@ describe('hover info', function() {
         });
     });
 
+    describe('hover info for negative data on a log axis', function() {
+        it('shows negative data even though it is infinitely off-screen', function(done) {
+            var gd = createGraphDiv();
+
+            Plotly.plot(gd, [{x: [1, 2, 3], y: [1, -5, 10]}], {
+                yaxis: {type: 'log'},
+                width: 500,
+                height: 400,
+                margin: {l: 0, t: 0, r: 0, b: 0}
+            })
+            .then(function() {
+                _hover(gd, 250, 200);
+                assertHoverLabelContent({
+                    nums: '\u22125', // unicode minus
+                    axis: '2'
+                });
+            })
+            .catch(fail)
+            .then(done);
+        });
+    });
+
+    describe('histogram hover info', function() {
+        it('shows the data range when bins have multiple values', function(done) {
+            var gd = createGraphDiv();
+
+            Plotly.plot(gd, [{
+                x: [0, 2, 3, 4, 5, 6, 7],
+                xbins: {start: -0.5, end: 8.5, size: 3},
+                type: 'histogram'
+            }], {
+                width: 500,
+                height: 400,
+                margin: {l: 0, t: 0, r: 0, b: 0}
+            })
+            .then(function() {
+                _hover(gd, 250, 200);
+                assertHoverLabelContent({
```

Comment:
Would mind checking that `plotly_hover` does emit the correct event data?

Reply:
Good catch -> tested in https://github.com/plotly/plotly.js/pull/2113/commits/2696c1c9ded8fb57c61ef103f08c374dda2a4f53